app/org/sagebionetworks/bridge/cache/ViewCache.java

```diff
@@ -79,11 +80,13 @@ public void setCacheProvider(CacheProvider cacheProvider) {
         return new ViewCacheKey<T>(RedisKey.VIEW.getRedisKey(id + ":" + clazz.getName()));
     }
     
-    private <T> String cacheView(ViewCacheKey<T> key, Supplier<T> supplier) throws JsonProcessingException {
+    private <T> String cacheView(ViewCacheKey<T> key, boolean useBridgeObjectMapper, int cachePeriod,
+            Supplier<T> supplier) throws JsonProcessingException {
         logger.debug("Caching JSON for " +key.getKey()+"'");
         T object = supplier.get();
-        String value = BridgeObjectMapper.get().writeValueAsString(object);
-        cache.setString(key.getKey(), value, BridgeConstants.BRIDGE_VIEW_EXPIRE_IN_SECONDS);
+        ObjectMapper mapper = (useBridgeObjectMapper) ? BridgeObjectMapper.get() : new ObjectMapper();
+        String value = mapper.writeValueAsString(object);
+        cache.setString(key.getKey(), value, cachePeriod);
```

Comment:
Unit tests?

Reply:
I revised this so the TTL and object mapper are injected Spring dependencies, which is more consistent with other configuration like this. Then added test specifically verifying the injected dependencies are being used.